libr/anal/esil.c

```diff
@@ -1824,6 +1825,39 @@ static bool esil_addeq(RAnalEsil *esil) {
 	return ret;
 }
 
+#if ESIL_MACRO
+static bool esil_inc_macro(RAnalEsil *esil) {
+	bool ret = false;
+	ut64 s;
+	char *src = r_anal_esil_pop (esil);
+	if (src && r_anal_esil_get_parm (esil, src, &s)) {
+		s++;
+		char *o = r_str_newf ("op:1,%s,+", src);
+		ret = r_anal_esil_push (esil, o);
+		free (o);
+	} else {
+		ERR ("esil_inc: invalid parameters");
+	}
+	free (src);
+	return ret;
+}
+
+static bool esil_inceq_macro(RAnalEsil *esil) {
+	bool ret = false;
+	ut64 s;
+	char *src = r_anal_esil_pop (esil);
+	if (src && r_anal_esil_get_parm (esil, src, &s)) {
+		s++;
+		char *o = r_str_newf ("op:1,%s,+,%s,=", src, src);
+		ret = r_anal_esil_push (esil, o);
```

Comment:
as long as this is not going to be recursive we can just add esil->expressed and use that in r_parse loop instead